lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -553,6 +565,7 @@ impl<'ty> ConstraintGroupBuilder<'ty> {
             state: into_inner!(state),
             close: into_inner!(close),
             address: into_inner!(address),
+            associated_token: if !is_init { associated_token } else { None },
```

Comment:
Mm, this is slightly odd I guess—we could have also done something more like what `ConstraintSeedsGroup` does with its `is_init` flag, but you have this mild illegal-but-seemingly-representable state where you have a ConstraintSeedsGroup with is_init = false embedded in a `ConstraintInitGroup`... meh, that's obviously not a big deal + that illegal state doesn't actually happen. Happy to switch to that style 🤷‍♂️